llvm/lib/Target/RISCV/RISCVRegisterInfo.td

```diff
@@ -191,6 +191,12 @@ def GPCR : RegisterClass<"RISCV", [CLenVT], 64, (add
       [RegInfo<64,64,64>, RegInfo<128,128,128>, RegInfo<64,64,64>]>;
 }
 
+class ConstrainedGPCR : RegisterOperand<GPCR> {}
+def TaggedGPCR : ConstrainedGPCR, MustBeTaggedCapabilityOp;
+def UnsealedGPCR : ConstrainedGPCR, MustBeUnsealedCapabilityOp;
+def ValidMutableGPCR : ConstrainedGPCR, MustBeUnsealedCapabilityOp, MustBeTaggedCapabilityOp;
+
+
```

Comment:
You have extra whitespace. But really, this doesn't belong in RegisterInfo, this belongs as special classes in InstructionInfo just like branch targets etc, that inherit from the relevant base classes.